src/scripts/main.js

```diff
@@ -1,5 +1,48 @@
 'use strict';
 
-document.addEventListener('click', (e) => {
-  // write code here
+document.addEventListener('DOMContentLoaded', () => {
+  const wall = document.querySelector('.wall');
+  const spider = document.querySelector('.spider');
+
+  // eslint-disable-next-line no-shadow
+  wall.addEventListener('click', (event) => {
+    const wallRect = wall.getBoundingClientRect();
+    const spiderRect = spider.getBoundingClientRect();
+
+    const spiderHalfWidth = spiderRect.width / 2;
+    const spiderHalfHeight = spiderRect.height / 2;
+    const borderWidth = 10;
+
+    // const clickX = event.clientX;
+    // const clickY = event.clientY;
+
+    const clickX = event.clientX - wallRect.left;
+    const clickY = event.clientY - wallRect.top;
+
+    let newLeft = clickX - spiderHalfWidth - borderWidth;
+    let newTop = clickY - spiderHalfHeight - borderWidth;
```

Comment:
The calculation for `newLeft` and `newTop` includes `borderWidth`, which might not be necessary here. This could lead to incorrect positioning of the spider within the wall. Consider removing `borderWidth` from these calculations unless it's intended for a specific reason.